### DISCUSSION TITLE
disable_selinux = true is not working

### DISCUSSION THREAD

**Discussion by un1xman** [mysticaltech/terraform-hcloud-kube-hetzner#1366] [Q&A]
As far as now, Selinux is working with enforcing mode, on nodes by default. After I change _disable_selinux=true_  it notaffected Selinux configs. Even, I destroy cluster and created with this config. Still selinux is active on nodes. Am I doing something wrong or incomplete?

**Comment by ghost** [comment]
Based on the [docs](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/docs/terraform.md), we have 2 ways to disable [SELinux](https://en.wikipedia.org/wiki/Security-Enhanced_Linux)

 1. We can disable selinux on all nodes using [disable_selinux](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/docs/terraform.md#input_disable_selinux) - `Disable SELinux on all nodes.`
    ```terraform
    disable_selinux = true
    ```

 2. We can disable selinux on agent nodes using [agent_nodepools](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/docs/terraform.md#input_agent_nodepools)
    ```terraform
    agent_nodepools = [
      {
        name        = "agent-small",
        server_type = "cx22",
        location    = "fsn1",
        selinux     = false
      }
    ```

But I see same result in bot cases
```shell
sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   enforcing
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Memory protection checking:     actual (secure)
Max kernel policy version:      33
```

**Comment by ghost** [comment]
Can we rely on [`preinstall_exec`](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/master/docs/terraform.md#input_preinstall_exec), to disable SELinux?
```yaml
  preinstall_exec = [
    "setenforce 0",
    "sed -i -E 's/^SELINUX=[a-z]+/SELINUX=disabled/' /etc/selinux/config"
  ]
```

**Comment by ghost** [comment]
Applied `preinstall_exec` from comment above and triggered nodes scaling

**`getenforce`**
```shell
Permissive
```

**`sestatus`**
```shell
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   permissive
Mode from config file:          disabled
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Memory protection checking:     actual (secure)
Max kernel policy version:      33
```

**`cat /etc/selinux/config`**
```shell
# This file controls the state of SELinux on the system.
# SELinux can be completly disabled with the "selinux=0" kernel
# commandline option.
#
# SELINUX= can take one of these three values:
#     enforcing - SELinux security policy is enforced.
#     permissive - SELinux prints warnings instead of enforcing.
SELINUX=disabled
# SELINUXTYPE= can take one of these three values:
#     targeted - Targeted processes are protected,
#     minimum - Modification of targeted policy. Only selected processes are protected.
#     mls - Multi Level Security protection.
SELINUXTYPE=targeted
```

#### And after manual node reboot

**`getenforce`**
```shell
Disabled
```

**`sestatus`**
```shell
SELinux status:                 disabled
```

**Comment by ghost** [comment]
[[Bug]: Disabling SELINUX option is not working #1370](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/issues/1370).
